seacrowd/sea_datasets/mywsl2023/mywsl2023.py

```diff
@@ -0,0 +1,181 @@
+
+    subset_id = _DATASETNAME
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{subset_id}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=subset_id,
+        )
+    ]
+
+    seacrowd_schema_config: list[SEACrowdConfig] = []
+
+    for seacrowd_schema in _SUPPORTED_SCHEMA_STRINGS:
+
+        seacrowd_schema_config.append(
+            SEACrowdConfig(
+                name=f"{subset_id}_{seacrowd_schema}",
+                version=SEACROWD_VERSION,
+                description=f"{_DATASETNAME} {seacrowd_schema} schema",
+                schema=f"{seacrowd_schema}",
+                subset_id=subset_id,
+            )
+        )
+
+    BUILDER_CONFIGS.extend(seacrowd_schema_config)
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "image_paths": datasets.Sequence(datasets.Value("string")),
+                    "texts": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{str(TASK_TO_SCHEMA[Tasks.SIGN_LANGUAGE_RECOGNITION]).lower()}":
+            features = schemas.image_text_features(label_names=_LABELS)
+
+        else:
+            raise ValueError(f"Invalid config: {self.config.name}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        split_generators = []
+
+        path = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+
+        for split in _SPLITS:
+            split_generators.append(
+                datasets.SplitGenerator(
+                    name=split,
+                    gen_kwargs={
+                        "path": os.path.join(path, "MyWSL2023 RAW DATA", _SPLIT_NAMES[split]),
+                    },
+                )
+            )
+
+        return split_generators
+
+    def _generate_examples(self, path: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        image_folder_paths = [os.path.join(path, folder) for folder in os.listdir(path)]
+
+        for idx, image_folder_path in enumerate(image_folder_paths):
+            image_paths = os.listdir(image_folder_path)
+
+            if self.config.schema == "source":
+                yield idx, {
+                    "image_paths": [os.path.join(image_folder_path, image_path) for image_path in image_paths],
+                    "texts": os.path.basename(image_folder_path),
+                }
+                idx += 1
```

Comment:
why do we need this `idx`?

Reply:
Since you already use enumerate, you don't need to increment the `idx` @Enliven26